pragma-node/src/handlers/entries/subscribe_to_entry.rs

```diff
@@ -0,0 +1,260 @@
+            }
+        } else {
+            let error_msg = "Could not serialize ack message.";
+            send_error_message(socket, error_msg).await;
+        }
+    } else {
+        let error_msg = "Invalid message type. Please check the documentation for more info.";
+        send_error_message(socket, error_msg).await;
+    }
+}
+
+/// Send the current median entries to the client.
+async fn send_median_entries(
+    socket: &mut WebSocket,
+    state: &AppState,
+    subscribed_pairs: &[String],
+) -> Result<(), EntryError> {
+    if subscribed_pairs.is_empty() {
+        return Ok(());
+    }
+    let entries = match get_subscribed_pairs_entries(state, subscribed_pairs).await {
+        Ok(response) => response,
+        Err(e) => {
+            send_error_message(socket, &e.to_string()).await;
+            return Err(e);
+        }
+    };
+    if let Ok(json_response) = serde_json::to_string(&entries) {
+        if socket.send(Message::Text(json_response)).await.is_err() {
+            send_error_message(socket, "Could not send prices.").await;
+        }
+    } else {
+        send_error_message(socket, "Could not serialize prices.").await;
+    }
+    Ok(())
+}
+
+/// Get the current median entries for the subscribed pairs and sign them as Pragma.
+async fn get_subscribed_pairs_entries(
+    state: &AppState,
+    subscribed_pairs: &[String],
+) -> Result<SubscribeToEntryResponse, EntryError> {
+    let median_entries =
+        get_current_median_entries_with_components(&state.timescale_pool, subscribed_pairs)
+            .await
+            .map_err(|e| e.to_entry_error(&subscribed_pairs.join(",")))?;
+
+    let mut response: SubscribeToEntryResponse = Default::default();
+    let now = chrono::Utc::now().timestamp();
+    for entry in median_entries {
+        let median_price = entry.median_price.clone();
+        let mut oracle_price: AssetOraclePrice = entry
+            .try_into()
+            .map_err(|_| EntryError::InternalServerError)?;
+
+        let signature = sign_median_price_as_pragma(
+            &state.pragma_signer,
+            &oracle_price.global_asset_id,
+            now as u64,
+            median_price,
+        )?;
+
+        oracle_price.signature = signature;
+        response.oracle_prices.push(oracle_price);
+    }
+    response.timestamp = now.to_string();
+    Ok(response)
+}
+
+/// Sign the median price as Pragma and return the signature
+/// 0x prefixed.
+fn sign_median_price_as_pragma(
+    signer: &SigningKey,
+    asset_id: &str,
+    timestamp: u64,
+    median_price: BigDecimal,
+) -> Result<String, EntryError> {
+    let hash_to_sign = get_entry_hash(PRAGMA_ORACLE_NAME, asset_id, timestamp, &median_price)
+        .map_err(|_| EntryError::InternalServerError)?;
+    let signature = signer
+        .sign(&hash_to_sign)
+        .map_err(EntryError::InvalidSigner)?;
+    Ok(format!("0x{:}", signature))
+}
+
+impl TryFrom<EntryComponent> for SignedPublisherPrice {
```

Comment:
move these to `types.rs` file

Reply:
Couldn't move them there cause it created circular dependencies - moved them to the entry repository